### PR TITLE
refactorings based on the feedback

### PR DESCRIPTION
**Pull request by vlas**

### CODE REVIEW

**Review by jpdolz — APPROVED** [review]
LGTM

**Review by marcserrat — APPROVED** [review]

File: connect/telegram_bot/bot_runner.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        )
+
+    def _get_settings(self):
+        try:
+            self.token = settings.TELEGRAM_BOT['TOKEN']
+            self.suffix = settings.TELEGRAM_BOT['COMMANDS_SUFFIX']
+            self.model_property = settings.TELEGRAM_BOT['HISTORY_LOOKUP_MODEL_PROPERTY']
+            self.conv_classes = settings.TELEGRAM_BOT['CONVERSATIONS']
+        except (KeyError, AttributeError, ImproperlyConfigured) as e:
+            exit(f'Error getting settings: {str(e)}')
```
Comment (ghost) [inline connect/telegram_bot/bot_runner.py]: ImproperlyConfigured is not raised here. I think is better to catch KeyError and AttributeError and raise ImproperlyConfigured with a meaningful message.
Probably is better to do so in the AppConfig class for your app (that is missed).

File: connect/telegram_bot/bot_runner.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+        if not self.token:
```
Comment (ghost) [inline connect/telegram_bot/bot_runner.py]: If you check the settings at application startup (i.e. in the AppConfig class), you don't need these checks.

File: connect/telegram_bot/bot_runner.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        if not isinstance(self.conv_classes, list):
+            sys.stderr.write('Conversation classes property is not a list')
+            sys.stderr.write('settings.TELEGRAM_BOT["CONVERSATIONS"]')
+            exit()
+
+    def handle(self, *args, **options):
+        updater = Updater(self.token)
+
+        for conversation in self.conv_classes:
+            sys.stdout.write(f'Setting up conversation handler {conversation} ..')
```
Comment (ghost) [inline connect/telegram_bot/bot_runner.py]: Why not use logger here?

File: connect/telegram_bot/telegram_conversation.py [hunk 4]
```diff
@@ -220,18 +212,14 @@ def _end_conversation(self, update):
...
 
     def execute_custom_command(self, update, command):
-        old_stdout = sys.stdout
-        sys.stdout = mystdout = StringIO()
-
+        out = StringIO()
         try:
-            execute_from_command_line(['', command])
-            self._reply(update, f'Result {command}:\n{mystdout.getvalue()}')
+            call_command(command, strerr=out, stdout=out)
```
Comment (ghost) [inline connect/telegram_bot/telegram_conversation.py]: Don't understand where if control that the command received from the telegram conversation is one of the declared available.
As we told, probably is not a good idea to expose all the available commands through the conversations (ex: createsuperuser).

**Review by ghost — CHANGES_REQUESTED** [review]

File: connect/telegram_bot/bot_runner.py (hunk 1 above)
Comment (vlas) [reply connect/telegram_bot/bot_runner.py]: fixed

File: connect/telegram_bot/bot_runner.py (hunk 2 above)
Comment (vlas) [reply connect/telegram_bot/bot_runner.py]: fixed

File: connect/telegram_bot/bot_runner.py (hunk 3 above)
Comment (vlas) [reply connect/telegram_bot/bot_runner.py]: changed

File: connect/telegram_bot/telegram_conversation.py (hunk 4 above)
Comment (vlas) [reply connect/telegram_bot/telegram_conversation.py]: added such check

**Review by ghost — APPROVED** [review]

**Pull request merged**